File: src/main/java/turniplabs/industry/block/tile/TileEntityRelay.java
package turniplabs.industry.block.tile;

import sunsetsatellite.energyapi.util.Connection;
import sunsetsatellite.energyapi.util.Direction;

public class TileEntityRelay extends TileEntityEnergyConductorDamageable{

    boolean isPowered = false;

    public TileEntityRelay(int capacity, int energy, int transfer) {
        setCapacity(capacity);
        setEnergy(energy);
        setTransfer(transfer);

        for (Direction dir : Direction.values()) {
            setConnection(dir, Connection.BOTH);
        }
    }

    public void setPowered(boolean powered) {
        isPowered = powered;
    }

    @Override
    public int receive(Direction dir, int amount, boolean test) {
        if (isPowered) return super.receive(dir, amount, test);
        return 0;
    }

    @Override
    public int provide(Direction dir, int amount, boolean test) {
        if (isPowered) return super.provide(dir, amount, test);
        return 0;
    }
}

File: src/main/kotlin/turniplabs/industry/block/tile/TileEntityRelay.kt
package turniplabs.industry.block.tile

import sunsetsatellite.energyapi.util.Connection
import sunsetsatellite.energyapi.util.Direction

class TileEntityRelay(capacity: Int, energy: Int, transfer: Int): TileEntityEnergyConductorDamageable() {

    private var isPowered: Boolean = false

    init {
        setCapacity(capacity)
        setEnergy(energy)
        setTransfer(transfer)

        for (dir: Direction in Direction.values())
            setConnection(dir, Connection.BOTH)
    }

    fun setPowered(powered: Boolean) {
        this.isPowered = powered
    }

    override fun receive(dir: Direction?, amount: Int, test: Boolean): Int {
        return if (isPowered) super.receive(dir, amount, test)
        else 0
    }

    override fun provide(dir: Direction?, amount: Int, test: Boolean): Int {
        return if (isPowered) super.provide(dir, amount, test)
        else 0
    }
}